src/main/scala/com/moesif/filter/MoesifApiFilter.scala

```diff
@@ -282,6 +282,10 @@ class MoesifApiFilter @Inject()(config: MoesifApiFilterConfig)(implicit mat: Mat
             }
           }
           def onFailure(context: HttpContext, ex: Throwable): Unit = {
+            val eventmodelMsg = "sendingEvents: " + sendingEvents.map(logEventModelHelper).mkString(",")
```

Comment:
@XinyuLuan , as a perf improvement to avoid unnecessary garbage collection, lets not build a string unless we need it. Meaning, lets move the string inside the "if" statement

Reply:
moved